x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -403,7 +411,7 @@ void start() {
                     }
                     ClusterState remoteClusterState = remoteClusterStateResponse.getState();
                     metadataVersion = remoteClusterState.metaData().version();
-                    autoFollowIndices(autoFollowMetadata, clusterState, remoteClusterState, patterns);
+                    autoFollowIndices(autoFollowMetadata, clusterState, remoteClusterState, patterns, thread);
```

Comment:
Do we need to handle the timeout case (line 409)? I know it always comes from a different thread.

Reply:
I don't think so because as you say, we are always on a callback thread with an already-truncated stack trace in that case.